traefik-letsencrypt-bookstack-docker-compose.yml

```diff
@@ -0,0 +1,114 @@
+version: "3.8"
+
+services:
+  traefik:
+    image: traefik:v3.0@sha256:e27e08111cced4a2505f7e86c9987cdc88e841e00cc04d628c5d9e1dcd75577c
+    command:
+      - "--log.level=WARN"
+      - "--accesslog=true"
+      - "--api.dashboard=true"
+      - "--api.insecure=true"
+      - "--ping=true"
+      - "--ping.entrypoint=ping"
+      - "--entryPoints.ping.address=:8082"
+      - "--entryPoints.web.address=:80"
+      - "--entryPoints.websecure.address=:443"
+      - "--providers.docker=true"
+      - "--providers.docker.endpoint=unix:///var/run/docker.sock"
+      - "--providers.docker.exposedByDefault=false"
+      - "--certificatesresolvers.letsencrypt.acme.tlschallenge=true"
+      # Email for Let's Encrypt (replace with your own e-mail)
+      - "--certificatesresolvers.letsencrypt.acme.email=example.mail@examplemailservice.com"
+      - "--certificatesresolvers.letsencrypt.acme.storage=/etc/traefik/acme/acme.json"
+      - "--global.checkNewVersion=true"
+      - "--global.sendAnonymousUsage=false"
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+      - /data/traefik/ssl:/etc/traefik/acme
+    ports:
+      - "80:80"
+      - "443:443"
+    healthcheck:
+      test: ["CMD", "wget", "http://localhost:8082/ping","--spider"]
+      interval: 10s
+      timeout: 5s
+      retries: 3
+      start_period: 5s
+    labels:
+      - "traefik.enable=true"
+      # Traefik URL (replace this with your own URL)
+      - "traefik.http.routers.dashboard.rule=Host(`example.url.com`)"
+      - "traefik.http.routers.dashboard.service=api@internal"
+      - "traefik.http.routers.dashboard.entrypoints=websecure"
+      - "traefik.http.services.dashboard.loadbalancer.server.port=8080"
+      - "traefik.http.routers.dashboard.tls=true"
+      - "traefik.http.routers.dashboard.tls.certresolver=letsencrypt"
+      - "traefik.http.services.dashboard.loadbalancer.passhostheader=true"
+      - "traefik.http.routers.dashboard.middlewares=authtraefik"
+      # Basic Authentication for Traefik Dashboard
+      # Username: admin (replace with your own username and encrypted password)
+      # Passwords must be encoded using MD5, SHA1, or BCrypt (check README.md on how to hash a password)
+      - "traefik.http.middlewares.authtraefik.basicauth.users=admin:$$2y$$10$$tzGrxnBWcBBCKpFWdvzJDOt/Y4ihpuxXuE9r7mP/eo1MRBGcmV9q2"
```

Comment:
Sorry to bother you, but it doesn't feel right to me to have passwords- even encrypted ones- in a docker-compose. The reasons might be subtle. I view docker-compose (or any other script) as a rather stable receipt the computer can execute. It shouldn't be touched if a password needs to be updated.
If you could add a `.env.example` file with the key for the password entry in it and a comment but no value, and then instructions how to copy and create that password. I want
- the user having to take action and create their own password, not take a public one because their lazy
- move the password out of the compose and just have compose reference the .env key. Now, I have no idea what you have to do to make compose read passwords from env- I hope that's somewhat straight forward? But you suggested that approach :)

Reply:
Sorry for the delay in this PR, due to holidays.

I've added a .env file and moved the basic_auth user and password to it. I've added a unencoded string to the password variable (won't work without encoding to bcrypt for example).

Now you have variables that in the docker-compose file that reference to the .env variables. I hope that everything is fine now. :)